src/pages/dar_application/ResearcherInfo_new.js

```diff
@@ -174,8 +213,26 @@ export default function ResearcherInfo(props) {
             type: 'button', // default button element type inside a form is "submit".
             className: 'button button-white btn-xs',
             style: { marginTop: 25 },
-            onClick: () => {}
-          }, ['Add Collaborator'])
+            onClick: () => addNewCollaborator(),
+            onChange: ({key: name, value,}) => formFieldChange({name, value}),
+          }, [
+            span({}, ['Add Collaborator'])
+          ]),
+
+          collaboratorsState
```

Comment:
It's cleaner to create a sub-component that will take the list of staff and perform the  template updates inside of it. You can see that approach with the `CollaboratorList` component that we currently use on the DAR Form.